astropy/table/table.py

```diff
@@ -2109,3 +2109,7 @@ def _is_mixin_column(self, col):
         If ``col`` is a string then it refers to a column name in this table.
         """
         return super(QTable, self)._is_mixin_column(col, quantity_is_mixin=True)
+
+    def __getstate__(self):
+        columns = dict((key, col.copy()) for key, col in self.columns.items())
```

Comment:
Try this:
```
columns = dict((key, col if isinstance(col, BaseColumn) else col_copy(col))
                for key, col in self.columns.items())
```
This *should* do two things (but not tested on my end): (1) not unnecessarily copy regular columns, (2) correctly copy SkyCoord.  `col_copy` is a mixin-safe version of column copying that should always work.